thunder/core/symbol.py

```diff
@@ -320,6 +321,20 @@ def __call__(self, *args, **kwargs):
             result = self.meta(*args, **kwargs)
             trace.pop_scope()
 
+        cd = get_compile_data()
+        if cd is not None and not cd.is_grad_enabled:
+            # If grad is disabled using `torch.no_grad` or `torch._C._set_grad_enabled(False)`,
+            # tag the results with `DETACHED_AUTOGRAD_GRAPH` which makes this Symbol a constant for
+            # vjp transform (applied later).
+            def tag_tensorproxy_output_as_detached(proxy):
+                if isinstance(proxy, TensorProxy):
+                    # We need to remove name from trace, otherwise replace will return a proxy with new name.
+                    trace.names.remove(proxy.name)
+                    return proxy.replace(tags=(ProxyTag.DETACHED_AUTOGRAD_GRAPH,))
```

Comment:
Is there a better way than this to enforce the same name is used for the output of `replace`

Reply:
Seems like there's no better way now. Proxies' init method calls 
https://github.com/Lightning-AI/lightning-thunder/blob/d60f85cf009793894e9329eca1ee543037237057/thunder/core/proxies.py#L149
which does not force provided name usage if it's already used (included in the `trace.names` set).

Reply:
Unfortunately, currently not. :/